pkg/collector/corechecks/system/wincrashdetect/probe/crashparse.go

```diff
@@ -0,0 +1,197 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+//go:build windows
+// +build windows
+
+package probe
+
+/*
+#cgo LDFLAGS: -l dbgeng -static
+#include "crashdump.h"
+*/
+import "C"
+import (
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// allow us to change for testing
+var readfn = doReadCrashDump
+
+type logCallbackContext struct {
+	loglines       []string
+	hasSeenRetAddr bool
+	unfinished     string
+}
+
+// maximum number of stack trace lines we'll look through, looking for non-"NT!" lines
+const maxLinesToScan = int(200)
+
+const (
+	bugcheckCodePrefix     = "Bugcheck code"
+	debugSessionTimePrefix = "Debug session time"
+	retAddrPrefix          = "RetAddr"
+	unableToLoadPrefix     = "Unable to"
+	ntBangPrefix           = "nt!"
+)
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+/*
+ * extra layer of indirection so that we can call the go parsing code
+ * (logLineCallbackGo) straight from the test function so that we can
+ * test out the parser with known input rather than actually calling
+ * the debugger
+ */
+//export logLineCallback
+func logLineCallback(voidctx C.PVOID, str C.PCSTR) {
+	ctx := (*logCallbackContext)(unsafe.Pointer(uintptr(voidctx)))
+	line := C.GoString(str)
+	logLineCallbackGo(ctx, line)
+}
+
+func logLineCallbackGo(ctx *logCallbackContext, line string) {
+	if !strings.Contains(line, "\n") {
+		ctx.unfinished = ctx.unfinished + line
+		return
+	}
+	if len(ctx.unfinished) != 0 {
+		line = ctx.unfinished + line
+		ctx.unfinished = ""
+	}
+	lines := strings.Split(line, "\n")
```

Comment:
suggestion: `bufio.Scanner` may be able to simplify some of this newline seeking logic for you.